pkgs/development/compilers/ghc/common-make-native-bignum.nix

```diff
@@ -386,7 +386,11 @@ stdenv.mkDerivation (
       # elimination on aarch64-darwin. (see
       # https://github.com/NixOS/nixpkgs/issues/140774 for details).
       ./Cabal-at-least-3.6-paths-fix-cycle-aarch64-darwin.patch
-    ];
+    ]
+
+    ++ lib.optionals (lib.versionAtLeast version "9.4") (
```

Comment:
Doesn't need to be a conditional anymore, because we only do 9.4 with `common-make-native-bignum.nix`.